src/main/java/core/basesyntax/data/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.data.strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
+        int newQuantity = currentQuantity - quantity;
+        if (newQuantity < 0) {
+            throw new RuntimeException("The resulting quantity of "
+                    + fruit + " cannot be negative.");
+        }
```

Comment:
The check for `newQuantity < 0` is redundant because the condition `currentQuantity < quantity` already throws an exception if there is not enough inventory. Consider removing this redundant check to simplify the code.